src/MainWindow.vala

```diff
@@ -817,4 +840,149 @@ public class PantheonCalculator.MainWindow : Gtk.ApplicationWindow {
+                return true;
+            case Gdk.Key.@8:
+            case Gdk.Key.KP_8:
+                button_8.activate();
+                return true;
+            case Gdk.Key.@9:
+            case Gdk.Key.KP_9:
+                button_9.activate();
+                return true;
+            case Gdk.Key.plus:
+            case Gdk.Key.KP_Add:
+                button_add.activate();
+                return true;
+            case Gdk.Key.minus:
+            case Gdk.Key.KP_Subtract:
+                button_sub.activate();
+                return true;
+            case Gdk.Key.asterisk:
+            case Gdk.Key.KP_Multiply:
+                button_mult.activate();
+                return true;
+            case Gdk.Key.slash:
+            case Gdk.Key.KP_Divide:
+                button_div.activate();
+                return true;
+            case Gdk.Key.period:
+            case Gdk.Key.decimalpoint:
+            case Gdk.Key.KP_Decimal:
+                button_point.activate();
+                return true;
+            case Gdk.Key.BackSpace:
+            case Gdk.Key.KP_Delete:
+                button_del.activate();
+                return true;
+            case Gdk.Key.Return:
+            case Gdk.Key.KP_Enter:
+            case Gdk.Key.KP_Equal:
+                button_calc.activate();
+                return true;
+        }
+
+        switch (keyval) {
+            case Gdk.Key.percent:
+                button_percent.activate();
+                return true;
+            case Gdk.Key.parenleft:
+                button_par_left.activate();
+                return true;
+            case Gdk.Key.parenright:
+                button_par_right.activate();
+                return true;
+        }
+
+        if ((mod_state & Gdk.ModifierType.CONTROL_MASK) == Gdk.ModifierType.CONTROL_MASK) {
+            switch (keyval) {
+                case Gdk.Key.X:
+                case Gdk.Key.x:
+                    button_clr.activate();
+                    return true;
+                case Gdk.Key.e:
+                    button_pow.activate();
+                    return true;
+                case Gdk.Key.p:
+                    button_pi.activate();
+                    return true;
+                case Gdk.Key.r:
+                    button_sr.activate();
+                    return true;
+                case Gdk.Key.C:
+                case Gdk.Key.c:
+                    if (entry.get_text () != "") {
+                        try {
+                            var output = eval.evaluate (entry.get_text (), decimal_places);
+                            clipboard.set_text (output);
+                        } catch (Core.OUT_ERROR e) {
+                                infobar_label.label = e.message;
+                                infobar.revealed = true;
+                        }
+                    }
+                    return true;
+                case Gdk.Key.V:
+                case Gdk.Key.v:
```

Comment:
#245

Reply:
https://github.com/elementary/calculator/pull/245 works when the display typed in to, But  when the display is read only, 245 need to be reworked with my code, plus im in the process of added control V support, so here what i'm using to read the clipboard        
 // TODO: Fix the to code paste number from clipboard
        var cancellable = new GLib.Cancellable();
        clipboard.read_async(new string[] {"text/plain;charset=utf-8"}, 0, cancellable, new OnClipboardTextReceivedDelegate(OnClipboardTextReceived)); 
        
//    TODO: Fix the to code paste number from clipboard
    private delegate void OnClipboardTextReceivedDelegate(GLib.Object source_object, GLib.AsyncResult res);


    private static void on_clipboard_text_received(Gdk.Clipboard clipboard, GLib.AsyncResult result, string? text) {
        try {
            //var output = eval.evaluate (text, decimal_places);
            //if (entry.get_text () != output) {
                //entry.set_text (output);
            //}
        } catch (Core.OUT_ERROR e) {
                //infobar_label.label = e.message;
                //infobar.revealed = true;
        }
    }